### PR TITLE
[1/n] upgrade react-native to 0.74

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Upgrade to react-native 0.74 for sdk 51
Closes ENG-1175

# How
A lot of the upgrade has been done so this was mostly getting expo go working. 

# Test Plan
- Expo Go + NCL

Bare expo is already using 0.74
Expo go will crash on iOS until #27780 is merged

### CODE REVIEW

File: apps/bare-expo/package.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     "@react-native-picker/picker": "2.5.1",
     "@react-native-segmented-control/segmented-control": "2.4.1",
     "@shopify/flash-list": "1.6.3",
+    "@shopify/react-native-skia": "0.1.221",
```
Comment (gabrieldonadel) [inline apps/bare-expo/package.json:48]: Is this change related to the update?

Comment (alanjhughes) [reply apps/bare-expo/package.json:48]: ios won't build if we don't add the podspec.json. Needed to add it here to generate it. Temporary solution. Probably need to add something to et

**Review by gabrieldonadel — APPROVED** [review]

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2465,11 +2465,275 @@
...
+					"-iframework",
+					"\"${PODS_ROOT}/../vendored/unversioned/@shopify/react-native-skia/libs/ios\"",
+					"$(inherited)",
+					"-DFOLLY_NO_CONFIG",
+					"-DFOLLY_MOBILE=1",
+					"-DFOLLY_USE_LIBCPP=1",
+					"-Wno-comma",
+					"-Wno-shorten-64-to-32",
+					"-DREACT_NATIVE_MINOR_VERSION=74",
+					"-DREANIMATED_VERSION=3.6.0",
```
Comment (Kudo) [inline apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj:2729]: please reset the file and edit again. i think this change was because you had expand the flags somehow.

File: yarn.lock [hunk 3]
```diff
@@ -18818,6 +18818,15 @@ string-width@^3.0.0, string-width@^3.1.0:
     is-fullwidth-code-point "^2.0.0"
     strip-ansi "^5.1.0"
 
+string-width@^4.1.0, string-width@^4.2.0, string-width@^4.2.3:
```
Comment (Kudo) [inline yarn.lock:18821]: does the yarn.lock intentional to change? may try remove node_modules and yarn from expo/expo root again.

File: apps/bare-expo/package.json (hunk 1 above)
Comment (Kudo) [reply apps/bare-expo/package.json:48]: i think you could remove this after you copy the podspec into expo-go

File: yarn.lock (hunk 3 above)
Comment (alanjhughes) [reply yarn.lock:18234]: Yep, I removed node modules and ran yarn again

File: apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj (hunk 2 above)
Comment (alanjhughes) [reply apps/expo-go/ios/Exponent.xcodeproj/project.pbxproj]: Fixed

**Review by lukmccall — APPROVED** [review]

**Review by Kudo — APPROVED** [review]
nit (you could do it on next upgrade)
squashing these two commits
- https://github.com/expo/react-native/commit/41eb880787e9335b8b3a3dd020ee05fdb4eb6070
- https://github.com/expo/react-native/commit/69ed4e7d5457e75e16e5cfa17ab2eeba903c770e

since prebuild hermes are not used now. we can also remove these commits
- https://github.com/expo/react-native/commit/361630c2c08ceea439b2a920270224ac3d9981dd (this commits include some unrelated changes as well)
- https://github.com/expo/react-native/commit/044e0ad6f23e09a54e776823bb0873a76681a996
- https://github.com/expo/react-native/commit/2c9cb737f4bd594f97e175b7e5053d15b6f524e3
